docs/Design_for_in-memory_storage_of_secure_CLI_properties.md

```diff
@@ -0,0 +1,613 @@
+# <span class="c25">Design for in-memory storage of secure Zowe CLI properties</span>
+
+<span class="c2 c1"></span>
+
+<span class="c9 c18 c19 c15">[Overview](#h.6jmu2spkqbf8)</span>
+
+<span class="c9 c15">[Use case](#h.aolhl4860uk3)</span>
+
+<span class="c9 c18 c19 c15">[Technical approach](#h.kprnl9ff3r1g)</span>
+
+<span class="c9 c18 c19 c15">[Security considerations](#h.wfxp6zit1nh7)</span>
+
+<span class="c9 c18 c19 c15">[In-memory secure properties in CI/CD pipelines](#h.3og4ocjuky49)</span>
+
+<span class="c9 c18 c19 c15">[Detailed design](#h.j5ioqvisua63)</span>
+
+<span class="c9 c18 c19 c15">[Pseudo code](#h.hz3fvdiwrwzf)</span>
+
+<span class="c9 c18 c19 c15">[Findings on available crypto libraries](#h.chrqtn7hchb)</span>
+
+<span class="c9 c18 c19 c15">[Proposed user documentation](#h.y21186yciq3d)</span>
```

Comment:
Should these links to section headers be updated to work in Markdown?

Reply:
This markdown was created by conversion tools (first to HTML and then to markdown), so it has many flaws. The large amount of embedded HTML/CSS make it difficult to maintain it in its markdown format. I am not sure whether we will make our future updates to this markdown, or to our original document and re-convert it. Therefore, I am reluctant to invest more effort fixing conversion shortcomings (I previously invested a lot of time on fixing formatting errors). In the case of this table of contents, I am more inclined to just delete the table of contents than to continually modify the TOC should we re-convert in the future.

Reply:
Fine with me 👍